Assets/Scripts/Possession.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Possession : MonoBehaviour
+{
+
+    private Player player;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        player = gameObject.GetComponent<Player>();   
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        
+    }
+
+    void OnTriggerStay(Collider other)
+    {
+        if(other.gameObject.tag == "Dead" && Input.GetKeyDown("space")){
+            Enemy deadEnemy = other.gameObject.GetComponent<Enemy>();
+            player.MaxHealth = deadEnemy.MaxHealth;
+            player.Health = deadEnemy.MaxHealth;
+
+
+            // Possess the enemy model
+            GameObject playerModel = transform.Find("Model").gameObject;
+            GameObject enemyModel = deadEnemy.transform.Find("Model").gameObject;
+            // Clone the enemy model within the player
+            GameObject newModel = GameObject.Instantiate(enemyModel, transform);
+            newModel.name = "Model";
+
+            // Possess the enemy weapon
+            player.SetWeapon(deadEnemy.weapon);

```

Comment:
You are setting the player's weapon to the enemy's weapon without actually transferring/cloning it. You later destroy the enemy object, which includes this weapon object. I think this may cause issues.

Reply:
Interesting. I didn't get any issues. What issues do you think would be caused?

Reply:
The problem is that you're assigning the player a reference to the enemy's weapon, not duplicating the enemy's weapon. Then, when you destroy the enemy GameObject, you're effectively destroying its weapon as well. So the player's weapon is now a null reference.

Reply:
Done!